rfcs/0000-observability-api.md

```diff
@@ -0,0 +1,223 @@
+          enableService: true 
+          port: 8080
+          deep: true
+        dns-operator:
+          enableService: true
+          port: 8080   
+
+    alerts:
+      namespace: my-amazing-namespace
+      component:
+        authorino:
+          operator-level: true
+          component-level: true
+        limitador:
+          operator-level: true
+          component-level: true  
+        kuadrant:
+          operator-level: true
+          component-level: true  
+
+          
+    dashboards:
+      namespace: my-amazing-namespace
+      component:
+        authorino:
+          operator-level: true
+          component-level: true
+        limitador:
+          operator-level: true
+          component-level: true  
+        kuadrant:
+          operator-level: true
+
+            
+```
+### Sample use case
+A use case a user might have would be they desire setting up tracing in the Limitador operator implementing the required endpoints and optional tag. The user also wants metrics setup with custom ports and requires service and serviceMonitors to be created for Kuadrant-operator and Authorino-operator as well as have the Authorino have a log level of Debug.
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: Observability
+spec:
+  tracing:  
+    component:
+      limitador:
+          endpoint: rpc://tempo.tempo.svc.cluster.local:4317
+          tags: tag1, tag2
+  metrics: 
+    component:
+      authorino: 
+        enableService: true
+        port: 8084
+        deep: true 
+      kuadrant:   
+        enableService: true
+        port: 8084 
+  logging:
+    component:
+      authorino: 
+        logLevel: debug      
+```
+### Status
+The status of the Observability CR will not be the observability stack is in a "healthy" state i.e Prometheus and Grafana is up and running. It should be the status of only the things that we contribute for example is new Logging and Tracing now in place or is the console plugin responding. We will not be taking responsibility for aspects we don't have control over. 
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+In terms of how the information supplied in the Observability CR will get passed to the other Kuadrant components, multiple options have been brought up:
+
+directly - like setting flags or configuration directly on the deployments of the different Kuadrant components.
+indirectly -  Passing the information to Authorino & limitador via the Authorino & Limitador CRs
+indirectly -  Passing the information to Authorino & limitador in the form of there own Observability CRs
+
+The best approach would be the indirect approach, meaning once the Observability CR is updated the information is passed to the relevant component CR. For example the tracing section in the Authorino CR spec would be updated with the required endpoints and other configuration in the Observability CR, this would then be updated in the Authorino CR. 
```

Comment:
Agreed.
The direct option could break easily when authorino or limitador change something in the deployment for example.
This does mean the available options that can be included in the Observability spec are limited to what's exposed via the Authorino & Limitador CRs (but changes can be requested)

Reply:
Yes that a good point to acknowledge